plugin/metadata/provider.go

```diff
@@ -48,6 +49,18 @@ type Provider interface {
 // Func is the type of function in the metadata, when called they return the value of the label.
 type Func func() string
 
+// IsLabel check that the provided name looks like a valid label name
+func IsLabel(label string) bool {
+	parts := strings.Split(label, "/")
```

Comment:
This allocates quite a bit of memory. `strings.Index` is a better choice I think.

also needs a test